src/history/hs_rec.c

```diff
@@ -156,13 +156,12 @@ __hs_insert_record_with_btree_int(WT_SESSION_IMPL *session, WT_CURSOR *cursor, W
  */
 static int
 __hs_insert_record_with_btree(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_BTREE *btree,
-  const WT_ITEM *key, const WT_UPDATE *upd, const uint8_t type, const WT_ITEM *hs_value,
-  WT_HS_TIME_POINT *stop_time_point, bool clear_hs)
+  const WT_ITEM *key, const uint8_t type, const WT_ITEM *hs_value,
+  WT_HS_TIME_POINT *start_time_point, WT_HS_TIME_POINT *stop_time_point)
```

Comment:
When we have an obsolete update in memory that we want to insert into the history store to allow older reads to read it we pass its start_time_point as WT_TS_NONE and its txnid. The original implementation of `__hs_insert_record_with_btree` didn't allow for that so I updated it to take a time point and NOT take the values off the update itself.